test/e2e/test/logstash/checks.go

```diff
@@ -83,29 +103,63 @@ func CheckStatus(b Builder, k *test.K8sClient) test.Step {
 }
 
 func (b Builder) CheckStackTestSteps(k *test.K8sClient) test.StepList {
-	println(test.Ctx().TestTimeout)
 	return test.StepList{
-		{
-			Name: "Logstash should respond to requests",
-			Test: test.Eventually(func() error {
-				client, err := NewLogstashClient(b.Logstash, k)
-				if err != nil {
-					return err
-				}
-				bytes, err := DoRequest(client, b.Logstash, "GET", "/")
-				if err != nil {
-					return err
-				}
-				var status logstashStatus
-				if err := json.Unmarshal(bytes, &status); err != nil {
-					return err
-				}
+		b.CheckMetricsRequest(k,
+			Request{
+				Name: "metrics",
+				Path: "/",
+			},
+			Want{
+				Status: "green",
+			}),
+		b.CheckMetricsRequest(k,
+			Request{
+				Name: "default pipeline",
+				Path: "/_node/pipelines/main",
+			},
+			Want{
+				Status: "green",
+				Match:  map[string]string{"pipelines.main.workers": "2"},
+			}),
+	}
+}
 
-				if status.Status != "green" {
-					return fmt.Errorf("expected green but got %s", status.Status)
+func (b Builder) CheckMetricsRequest(k *test.K8sClient, req Request, want Want) test.Step {
+	return test.Step{
+		Name: fmt.Sprintf("Logstash should respond to %s requests", req.Name),
+		Test: func(t *testing.T) {
```

Comment:
I think we should change this to a `test.Eventually` block, so it will be retried if the endpoint isn't up and running straight away

```
		Test: test.Eventually(func() error {

			// send request and parse to map obj
			client, err := NewLogstashClient(b.Logstash, k)
			if err != nil {
				return err
			}

			bytes, err := DoRequest(client, b.Logstash, "GET", req.Path)
			if err != nil {
				return err
			}

			var response map[string]interface{}
			err = json.Unmarshal(bytes, &response)
			if err != nil {
				return err
			}

			// parse response to ucfg.Config for traverse
			res, err := settings.NewCanonicalConfigFrom(response)
			if err != nil {
				return err
			}

			// check status
			status, err := res.String("status")
			if err != nil {
				return err
			}
			if status != want.Status {
				fmt.Errorf("expected %s but got %s", want.Status, status)
			}

			// check expected string
			for k, v := range want.Match {
				str, err := res.String(k)
				if err != nil {
					return err
				}
				if str != v {
					fmt.Errorf("expected %s to be %s but got %s", k, v, str)
				}
			}
			return nil
		}),
```